interview-questions.md

```diff
@@ -7,63 +7,72 @@ First, without external resources. Challenge yourself to answer from memory as i
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
 1. What is the difference between a parameter and an argument?
+   - Answer:
+      > The difference between a parameter and an argument is that a parameter is a placeholder in the creation of a function and an argument is data passed into a function where a parameter would then be fulfilled.
 
-  Your answer:
-
-  Researched answer:
-
-
+   - Researched Answer:
+      > The difference between a parameter and an argument is that parameters are the named placeholders for a value while creating functions. Once the function is created, the data that is passed into the function call would then be considered the function argument.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
+   - Answer:
+      > The JavaScript built in method takes in value, index, and array. While they are all useful in different situations, value is the only required parameter out of the three. The index and array can be useful when finding an index of a value or returning full arrays are necessary. I recently realized that the exact terms aren't required to equal value, index, and array.
 
-  Your answer:
-
-  Researched answer:
-
-
+   - Researched Answer:
+      > The JavaScript built in map method takes in one required parameter, a callback function. The callback function is made up of the value, index and array parameters, of which the value is the only required to build the callback function. The map method can take in a second argument thisArgs, which can be used to reference the "this" reference of internal properties values of the callback function. If thisArgs isn't used, the predefined value of undefined will be used.
```

Comment:
Awesome work! 
To jump a bit into your question re: thisArgs with my current understanding. 
Map is a method that belongs to the class of Array.prototype in JS. Prototype is a whole other hour long conversation. 'This' here refers to the parent object that the map is being called upon and all of the attributes available to it as part of 'Array.prototype.this' (this being the current object). 

The use case for this argument are few and far between, but most commonly used in React.